packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -61,120 +61,96 @@ export async function makeWallet(
     inboxStateChangeHandler(getInboxState());
   }
 
-  function checkOrder(a0, a1, b0, b1) {
-    if (a0 === b0 && a1 === b1) {
-      return true;
-    }
-
-    if (a0 === b1 && a1 === b0) {
-      return false;
-    }
-
-    throw new TypeError('Cannot resolve assay ordering');
-  }
-
   async function makeOffer(date) {
     const {
-      meta: { purseName0, purseName1, instanceId },
+      meta: { instanceId, ...rest },
       offerRules,
     } = dateToOfferRec.get(date);
 
-    const purse0 = petnameToPurse.get(purseName0);
-    const purse1 = petnameToPurse.get(purseName1);
-
-    const {
-      payoutRules: [
-        {
-          units: { extent: extent0, assayId: assayId0 },
-        },
-        {
-          units: { extent: extent1, assayId: assayId1 },
-        },
-      ],
-    } = offerRules;
-
-    const instanceHandleP = E(registrar).get(instanceId);
-    const regAssay0P = E(registrar).get(assayId0);
-    const regAssay1P = E(registrar).get(assayId1);
-    const purseAssay0P = E(purse0).getAssay();
-    const purseAssay1P = E(purse1).getAssay();
-    const purseUnit0P = E(purseAssay0P).makeUnits(extent0 || 0);
-    const purseUnit1P = E(purseAssay0P).makeUnits(extent1 || 0);
+    // Collapse purseName0, purseName1, ... into purseNames
+    const purseNames = [];
+    for (let i = 0; i < offerRules.payoutRules.length; i += 1) {
+      const purseName = rest[`purseName${i}`];
+      if (purseName === undefined) {
+        break;
+      }
+      purseNames.push(purseName);
+    }
+    const rawPurses = purseNames.map(purseName =>
+      petnameToPurse.get(purseName),
+    );
 
-    const [
-      instanceHandle,
-      regAssay0,
-      regAssay1,
-      purseAssay0,
-      purseAssay1,
-      purseUnit0,
-      purseUnit1,
-    ] = await Promise.all([
-      instanceHandleP,
-      regAssay0P,
-      regAssay1P,
-      purseAssay0P,
-      purseAssay1P,
-      purseUnit0P,
-      purseUnit1P,
-    ]);
+    const { payoutRules } = offerRules;
 
-    // =====================
-    // === AWAITING TURN ===
-    // =====================
+    const instanceP = E(registrar)
+      .get(instanceId)
+      .then(instanceHandle => E(zoe).getInstance(instanceHandle));
+    const regAssaysAP = payoutRules.map(({ units: { assayId } }) =>
```

Comment:
What is the suffix AP intended to indicate? I'm happy to adopt a new convention, but I'd like to know what it is.

Reply:
Array of Promises.